### DISCUSSION TITLE
我们有需求基于富文本做一个pdf内容模板编辑器，主要用于将html转换为pdf

### DISCUSSION THREAD

**Discussion by zhaoqiang1024** [wangeditor-next/wangEditor-next#373] [General]
## 功能描述
我是一个后端，有需求基于富文本做一个pdf内容模板编辑器主要用于将html转换为pdf，基于我们现有功能我们可以把html转换为pdf，整体功能需要几个功能：
1、基于样式调整（这个目前已经可以支持了）。
2、checkbox选择框选项。这个功能目前可以使用待办来实现，但有个问题是待办只一行只能有一个。
3、需要增加pdf强制分页，需要基于某些样式目前在现有模板中已经实现但编辑器不支持。
4、需要一个可以在范围内随意拖动的文本框，目前在现有模板中实现，但编辑器不知该怎么处理。

## 提炼几个功能点
- 功能1 希望可以出一个与待办同类型的checkbox按钮但区别于待办，该按钮可以在同一行中随意添加。
- 功能2 有没有类似参考可以支持thymeleaf语法。
- 功能3 希望可以支持input框这些，同checkbox按钮同一行可以放置多个

有大佬可以帮粗略评估一下增加拖动文本框的可行性和复杂度吗，从api上来看，增加checkbox框不是很难，thymeleaf语法暂时不支持也可以，但是拖动文本框这个是需要尽快开发，但我是个后端对前端技术一知半解有些东西看的不很懂，希望前端大佬们给一个复杂度和未来可能出现的问题
## 原型图

*涉及到 UI 改动的功能，请一定提供原型图。原型图能表明功能即可，不要求规范和美观*

## 可参考的案例

*是否已有可参考的案例（如其他编辑器），有的话请给出链接*

**Comment by cycleccc** [comment]
拖动文本框这个 plate 有做，自己实现一个会比较麻烦

**Comment by zhaoqiang1024** [comment]
> 拖动文本框这个 plate 有做，自己实现一个会比较麻烦

确实是比较麻烦，我们目前生成模板都是后端来出html，涉及到改样式之类的都要跟研发来沟通，无法达到产品自主运营的效果。plate这个只是单行进行拖动，而我要的效果文本框不是一个块级结构，而是可以在页面上随意拖动，基于position+z-index来控制它的层级

**Comment by zhaoqiang1024** [comment]
> 拖动文本框这个 plate 有做，自己实现一个会比较麻烦

我是一个后端前端技术比较薄弱，有一些东西看不明白，之前有一些比较小的私人客户来找想做类似的功能都被我拒绝了。我尝试根据文档开发一些小的组件出来，但是涉及到复杂交互的就搞不明白了

**Comment by zhaoqiang1024** [comment]
另外，我发现表格一个交互上的总是，默认的表格通过getHtml获取以后是不带样式的，但是在editor里是带样式的，而单元格边框的按钮选项又只能一个一个单元格设置不能选中设置，这个是不是要优化一下

**Comment by cycleccc** [comment]
> 另外，我发现表格一个交互上的总是，默认的表格通过getHtml获取以后是不带样式的，但是在editor里是带样式的，而单元格边框的按钮选项又只能一个一个单元格设置不能选中设置，这个是不是要优化一下

这个设计是因为获取的 html 是可以完全自定义的，而 editor 内的则是带了默认样式，也可以修改，通过 id 属性定位修改即可。
![image](https://github.com/user-attachments/assets/10b98b52-6f5e-4e72-8ca1-dee6792a2274)

多选设置单元格样式确实可以优化，这个后续可以做。

**Comment by cycleccc** [comment]
> > 拖动文本框这个 plate 有做，自己实现一个会比较麻烦
> 
> 确实是比较麻烦，我们目前生成模板都是后端来出html，涉及到改样式之类的都要跟研发来沟通，无法达到产品自主运营的效果。plate这个只是单行进行拖动，而我要的效果文本框不是一个块级结构，而是可以在页面上随意拖动，基于position+z-index来控制它的层级

页面上随意拖动没太懂，还以为 plate 那样已经够用了。
